src/servers/src/mysql/handler.rs

```diff
@@ -364,6 +386,116 @@ impl<W: AsyncWrite + Send + Sync + Unpin> AsyncMysqlShim<W> for MysqlInstanceShi
         let _timer = crate::metrics::METRIC_MYSQL_QUERY_TIMER
             .with_label_values(&[crate::metrics::METRIC_MYSQL_TEXTQUERY, db.as_str()])
             .start_timer();
+
+        let query_upcase = query.to_uppercase();
+        if query_upcase.starts_with("PREPARE ") {
+            let mut outputs = vec![];
+            match ParserContext::parse_mysql_prepare_stmt(query, query_ctx.sql_dialect()) {
+                Ok((stmt_name, stmt)) => {
+                    let prepare_results =
+                        self.do_prepare(&stmt, query_ctx.clone(), stmt_name).await;
+                    match prepare_results {
+                        Ok(_) => {
+                            outputs.push(Ok(Output::new_with_affected_rows(0)));
+                            writer::write_output(writer, query_ctx, outputs).await?;
+                            return Ok(());
+                        }
+                        Err(e) => {
+                            writer
+                                .error(ErrorKind::ER_SP_BADSTATEMENT, e.output_msg().as_bytes())
+                                .await?;
+                            return Ok(());
+                        }
+                    }
+                }
+                Err(e) => {
+                    writer
+                        .error(ErrorKind::ER_PARSE_ERROR, e.output_msg().as_bytes())
+                        .await?;
+                    return Ok(());
+                }
+            }
+        } else if query_upcase.starts_with("EXECUTE ") {
+            match ParserContext::parse_mysql_execute_stmt(query, query_ctx.sql_dialect()) {
+                // TODO: similar to on_execute, refactor this
+                Ok((stmt_name, params)) => {
+                    let sql_plan = match self.plan(stmt_name) {
+                        None => {
+                            writer
+                                .error(
+                                    ErrorKind::ER_UNKNOWN_STMT_HANDLER,
+                                    b"prepare statement not found",
+                                )
+                                .await?;
+                            return Ok(());
+                        }
+                        Some(sql_plan) => sql_plan.clone(),
```

Comment:
Since `self.plan(stmt_name)` returns an owned `sql_plan`, there seems no need to clone it again.